src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/ImmutableDispatchRuntime.cs

```diff
@@ -478,8 +478,25 @@ internal DispatchOperationRuntime GetOperation(ref Message message)
             return _demuxer.GetOperation(ref message);
         }
 
+        private void ReceiveContextRPCFacet_CreatIfRequired_Shim(MessageRpc rpc)
+        {
+            // see https://referencesource.microsoft.com/#System.ServiceModel/System/ServiceModel/Dispatcher/ReceiveContextRPCFacet.cs,25
+            ReceiveContext receiveContext = null;
+            if (!ReceiveContext.TryGet(rpc.Request, out receiveContext))
+            {
```

Comment:
We shouldn't be throwing if the receive context is missing. That just means the transport for this request doesn't use it. This is what's causing the tests to fail. If there isn't one, then anything involving the receive context becomes a no-op

Reply:
done. Thanks!